docs/contrib/releases.adoc

```diff
@@ -0,0 +1,88 @@
+== SDK Release Documentation
+
+The release process results in new versions of the core and associated SDKs being published to Maven. This docmument should help you understand how releases work and how to create one.
+
+=== How Releases Work
+
+There are a number of stages in the process. Some are automated and some are manual. At a high level, releases work as follows:
+
+* The release artefacts are created and pushed to link:https://oss.sonatype.org/#stagingRepositories[Nexus]. 
+* The developer logs into Nexus and creates a temporary repository where the release artefacts can be downloaded and tested (if necessary).
+* The developer then uses Nexus to push the release artefacts to Maven.
+
+The initial step where the release artefacts are built and pushed to Nexus is automated in CircleCI
+
+=== CircleCI Release Job
+
+
+CircleCI is responsible for the initial build and push to Nexus. CircleCI notices a new release being created and kicks off a special release workflow. The following steps take place:
+
+* A standard build job runs, i.e. the code is built and tested.
+* If the build is successful then a release job runs.
+  ** The `VERSION_NAME` and `VERSION_CODE` in `gradle.properties` are validated.
+  ** The release tag is validated to make sure it matches the version number in `gradle.properties`
+  ** The core and associated SDKs are released to link:https://oss.sonatype.org/#stagingRepositories[Nexus.]
+
+A small amount of preparation needs to take place before this process kicks off. The version number must be updated in `gradle.properties`
+
+=== How to Prepare a Release
+
+Ensure the git repo is clean and there are no outstanding changes. Then check out a new branch.
+
+[source, bash]
+--
+git checkout -b release-x.y.z
+--
+
+Update the `VERSION_NAME` property in the `gradle.properties` file at the root of the repository. The `VERSION_NAME` **must be in the format x.y.z or x.y.z-label**
```

Comment:
What about: The `VERSION_NAME` must follow the [Semantic Versioning](https://semver.org/)